controllers/modInfo.js

```diff
@@ -0,0 +1,134 @@
+              return res.render('module', { mods: docs, exists: false, mod: { model: '', shape: '', orientation: '', notes: '', flipped: false, locationCode: ''}, plants: plant, x, y, tag: '' });
+            }
+            IndividualPlant
+              .find({ module: module._id })
+              .exec((err, plantedplants) => {
+                if (err) { return next(err); }
+                if (!plantedplants) {
+                  req.flash('info', { msg: 'Editing Empty Module' });
+                  return res.render('module', { plants: plant, x, y });
+                }
+                req.flash('info', { msg: 'Editing Module' });
+                console.log('plantedPlants', plantedplants);
+
+                return res.render('modinfo', { mods: docs, exists: true, mod: module, plants: plant, x, y, plantedPlants: plantedplants });
+              })
+          });
+      });
+    })
+};
+//5f053deec74c463ee1d67760 {$in : tags}
+exports.updateModInfo = (req, res, next) => {
+let tag = req.query.tags;
+if (!tag) {
+  console.log('No tags selected in search');
+}
+  Mod
+    .find({tag })
+    .distinct('_id')
+    .exec((err, usedMods) => {
+      if (err) { return next(err); }
+      let tagMods =  new Array();
+      console.log(tagMods);
+      let search = '_id'
+      for (let i = 0; i < usedMods.length; i++) {
+        if (usedMods[i]) {
+          let search = 'module';
+          tagMods.push(mongoose.Types.ObjectId(usedMods[i]));
+        }
+        else {
+          tagMods = {};
+        }
+      }
+      console.log('tagmods',tagMods);
+      IndividualPlant
+        .aggregate([
+  {
+  $match : { 'module': { $in: tagMods}}
+},
+		{"$group" : {_id:"$plant", count:{$sum:1}}},
+    {
+     $lookup: {
+         from: 'plants',
+         localField: '_id',
+         foreignField: "_id",
+         as: "plants"
+     },
+ },
+ {
+  $project : {
+    _id:1,
+    count:1,
+    plantName : '$plants.scientificName',
+    commonName : '$plants.commonName',
+    module : '$module._id'
+  }
+}
+ // { "$unwind": { "path" : "$_id" } },
+]).exec((err, data) => {
+    if (err) { return next(err);}
+    //console.log(JSON.stringify(data));
+    console.log('usedMods', usedMods);
+    return res.send(data);
+   });
+});
+};
+
+exports.updateModInfo1 = (req, res, next) => {
+  // let x = req.params.x
+  // let y = req.params.y
+            IndividualPlant
+              .find()
+              // .distinct('plant')
+              // .group({
+              //   key:{'plant'},
+              //   reduce:function(cur,result){result.count+=1},
+              //   initial:{count:0}
+              // })
+              // .aggregate(aggregatorOpts)
+              .exec((err, plantedplants) => {
+                if (err)
+                console.log('plantedPlants', plantedplants);
+                return res.send( {plantedPlants: plantedplants });
+});
+};
+
+exports.updateModInfo2 = (req, res, next) => {
```

Comment:
What is the purpose of the two different update methods?

Reply:
Actually nothing. I will take out